### PR TITLE
dx12/gl/vk: Rendering fixes; surface format enhancements and bugfixes for vulkan

### PR DESCRIPTION
**Pull request by kd-11**
Blade Arcus from Shining EX issues a clear command before selecting a surface target at the very begining of the game. https://github.com/RPCS3/rpcs3/issues/1681

### CODE REVIEW

**Comment by kd-11** [comment]
I've included the surface format fixes here. Currently only for opengl, but i'll look at the other backends soon.

**Comment by SakataGintokiYT** [comment]
@kd-11 
-Fixed blur  / Vulkan 
-Color is incorrect 

Fotoszok.pl
Fotoszok.pl
Fotoszok.pl
Fotoszok.pl
Fotoszok.pl
Fotoszok.pl

**Comment by kd-11** [comment]
I've noticed the color corruption issue. Fixing...

**Comment by kd-11** [comment]
It seems channel remapping is ignored when sampling rtts. I've rewritten the surface formats to also return a native layout mapping for use when a default component mapping is desired.

**Comment by SakataGintokiYT** [comment]
-Fixed color 
Fotoszok.pl
Fotoszok.pl

**Comment by raven02** [comment]
The DBZ shadow issue probably same as Hitman that surround by black textures .I remember @vlj  mentioned before it may be sampler parameter issue.

- Fix Vulkan shadow in Hitman 2. For some reasons with Vulkan Hitman 2 shadows appear correctly but are surrounded by a black texture. I guess it's a sampler parameter not properly set but I can be wrong.

**Comment by kd-11** [comment]
@SakataGintokiYT I've included a fix for the shadows in hitman. Please check if it breaks anything.

**Comment by SakataGintokiYT** [comment]
@kd-11 
Fotoszok.pl

**Comment by raven02** [comment]
Looks like no shadow.

**Comment by kd-11** [comment]
@SakataGintokiYT That is a different bug actually. If you look at the previous screenshot, there is a black "shadow" around the characters but no actual character shadows. I'll look into the missing shadows separately as i dont want to bloat this pull request

@Nekotekina Please review

File: rpcs3/Emu/RSX/VK/VKVertexProgram.cpp [hunk 1]
```diff
@@ -125,9 +125,13 @@ static const reg_info reg_table[] =
 	{ "gl_ClipDistance[1]", false, "dst_reg5", ".z", false },
 	{ "gl_ClipDistance[2]", false, "dst_reg5", ".w", false },
 	{ "gl_PointSize", false, "dst_reg6", ".x", false },
+
+	/*Disable user clip planes until they are properly handled
```
Comment (Nekotekina) [inline rpcs3/Emu/RSX/VK/VKVertexProgram.cpp]: Use // comments plz.

**Comment by kd-11** [comment]
@Nekotekina updated and rebased

**Comment by kd-11** [comment]
Travis-ci is crashing due to apt error

**Comment by Nekotekina** [comment]
Something is wrong with both travis and appveyor... I guess we need to wait.

**Comment by Nekotekina** [comment]
Try.

**Comment by raven02** [comment]
@kd-11 .Try rebase and should be ok for Travis.

**Pull request merged**